node/src/accountant/scanners/payable_scanner/finish_scan.rs

```diff
@@ -0,0 +1,249 @@
+        let mut subject = PayableScannerBuilder::new()
+            .sent_payable_dao(sent_payable_dao)
+            .failed_payable_dao(failed_paybale_dao)
+            .build();
+        subject.mark_as_started(SystemTime::now());
+        let response_skeleton = ResponseSkeleton {
+            client_id: 1234,
+            context_id: 5678,
+        };
+        let sent_payables = SentPayables {
+            payment_procedure_result: Ok(BatchResults {
+                sent_txs: sent_txs.clone(),
+                failed_txs: failed_txs.clone(),
+            }),
+            payable_scan_type: PayableScanType::Retry,
+            response_skeleton_opt: Some(response_skeleton),
+        };
+        let logger = Logger::new(test_name);
+
+        let result = subject.finish_scan(sent_payables, &logger);
+
+        let sent_payable_insert_new_records_params =
+            sent_payable_insert_new_records_params_arc.lock().unwrap();
+        let failed_payable_update_statuses_params =
+            failed_payable_update_statuses_params_arc.lock().unwrap();
+        assert_eq!(sent_payable_insert_new_records_params.len(), 1);
+        assert_eq!(
+            sent_payable_insert_new_records_params[0],
+            sent_txs.iter().cloned().collect::<BTreeSet<_>>()
+        );
+        assert_eq!(failed_payable_update_statuses_params.len(), 1);
+        let updated_statuses = failed_payable_update_statuses_params[0].clone();
+        assert_eq!(updated_statuses.len(), 2);
+        assert_eq!(
+            updated_statuses.get(&make_tx_hash(10)).unwrap(),
+            &FailureStatus::RecheckRequired(Waiting)
+        );
+        assert_eq!(
+            updated_statuses.get(&make_tx_hash(20)).unwrap(),
+            &FailureStatus::RecheckRequired(Waiting)
+        );
+        assert_eq!(
+            result,
+            PayableScanResult {
+                ui_response_opt: Some(NodeToUiMessage {
+                    target: MessageTarget::ClientId(response_skeleton.client_id),
+                    body: UiScanResponse {}.tmb(response_skeleton.context_id),
+                }),
+                result: NextScanToRun::PendingPayableScan,
+            }
+        );
+        let tlh = TestLogHandler::new();
+        tlh.exists_log_matching(&format!(
+            "WARN: {test_name}: While retrying, 2 transactions with hashes: {} have failed.",
+            join_with_separator(failed_txs, |failed_tx| format!("{:?}", failed_tx.hash), ",")
+        ));
+        tlh.exists_log_matching(&format!(
+            "INFO: {test_name}: The Payables scan ended in \\d+ms."
+        ));
+    }
+
+    #[test]
+    fn payable_scanner_with_error_works_as_expected() {
+        execute_payable_scanner_finish_scan_with_an_error(PayableScanType::New);
+        execute_payable_scanner_finish_scan_with_an_error(PayableScanType::Retry);
+    }
+
+    fn execute_payable_scanner_finish_scan_with_an_error(payable_scan_type: PayableScanType) {
```

Comment:
it is a tradition that we begin names of this kind of functions with the word `test_`